src/convert.cpp

```diff
@@ -113,26 +113,27 @@ ddwaf_object* to_ddwaf_object(
   }
   if (val.IsNumber()) {
     mlog("creating Number");
-    return ddwaf_object_signed(object, val.ToNumber().Int64Value());
+    if (coerceToString) {
+      return ddwaf_object_string(object, std::to_string(val.ToNumber().Int64Value()).c_str());
```

Comment:
why do we need the `std::to_string()` here but not elsewhere ?

Reply:
also can't we just coerce the values to string with `Napi::String Napi::Value::ToString() const;` for every different value like specified [here](https://github.com/nodejs/node-addon-api/blob/main/doc/value.md#tostring)

Reply:
if just doing Value::ToString work, we can skip the whole type checking and do a unviersal ToStrin() no ?

Reply:
This way, `NaN` and `Infinity` will be parsed to its string representation ("NaN", "Infinity") instead of the one expected in the tests ("0") that is the outcome from the `ToNumber().Int64Value()`. It will also impact to floats (for 42.42 input the test expects a string with the integer part "42").

Reply:
In this case the tests I made were to check non-regression, they're not the spec. So if we add support for floats NaN and Infinity now, which I think we can, we can also update the tests to reflect that.

What do you think about converting NaN, Infinity and floats to their string counterparts ?

Reply:
I think it's more accurate and represents their intrinsic value, at least more than turning them to `"0"`.